examples/secret.py

```diff
@@ -0,0 +1,98 @@
+import typing
+
+import discord
+from discord.ext import commands
+
+bot = commands.Bot(command_prefix=commands.when_mentioned, description="Nothing to see here!")
+
+@bot.event
+async def on_command(ctx):
+    try:
+        await ctx.message.delete(delay=5)
+    except discord.HTTPException:
+        # Oh well, looks like we don't have the ability
+        # to hide the secret.
+        pass
+
+# the `hidden` keyword argument hides it from the help command. 
+@bot.group(invoke_without_command=True, hidden=True)
+async def secret(ctx: commands.Context):
+    """What is this "secret" you speak of?"""
+    if ctx.invoked_subcommand is None:
+        await ctx.send('Shh!', delete_after=5)
+
+def create_overwrites(ctx, *objects):
+    """This is just a helper function that creates the overwrites for the 
+    voice/text channels.
+
+    A `discord.PermissionOverwrite` allows you to determine the permissions
+    of an object, whether it be a `discord.Role` or a `discord.Member`.
+
+    In this case, the `read_messages` permission is being used to hide the channel
+    from being viewed by whoever does not meet the criteria, thus creating a
+    secret channel.
+    """
+
+    overwrites = {
+        obj: discord.PermissionOverwrite(read_messages=True) for obj in objects
+    }
+ 
+    overwrites.setdefault(ctx.guild.default_role, discord.PermissionOverwrite(read_messages=False))
+    # prevents the default role (@everyone) from viewing the channel
+    # if it isn't already allowed to view the channel.
+    
+    overwrites[ctx.guild.me] = discord.PermissionOverwrite(read_messages=True)
+    # makes sure the client is always allowed to view the channel.
+
+    return overwrites
+
+@commands.guild_only()
+@secret.command()
+async def text(ctx: commands.Context, name: str, *objects: typing.Union[discord.Role, discord.Member]):
+    """This may or may not make a text channel with a specified name 
```

Comment:
This type of comment is cheeky but I think it can also be confusing to a reader. Best to just be explicit when explaining.